### PR TITLE
feat(tester): add job for running docker-compose integration tests

### PR DESCRIPTION
**Pull request by amartsinovich-dialpad**
Surprisingly for me, docker-compose works pretty seamlessly in circleci.

Green build: https://app.circleci.com/pipelines/github/talkiq/realtime/1883/workflows/a291ae1f-9dc3-4a6e-8cc6-bffad54e307c/jobs/9329

### CODE REVIEW

File: tester/orb.yaml [hunk 1]
```diff
@@ -134,3 +134,47 @@ jobs:
...
+      args:
+        default: ''
+        description: |
+          Arguments to pass to pytest
+        type: string
+      cwd:
+        default: '.'
+        description: >
+          Working directory used when running pip / pytest commands.
+        type: string
```
Comment (jonathan-johnston) [inline tester/orb.yaml:154]: Here too, is this job related to `pytest`, per se? Can you run any integration test within this job independent of `pytest`?

File: tester/orb.yaml [hunk 2]
```diff
@@ -134,3 +134,47 @@ jobs:
...
+      Runs integration tests using docker-compose
+    docker:
+      - image: <<parameters.executor>>
+    resource_class: <<parameters.resource_class>>
+    parameters:
+      args:
+        default: ''
+        description: |
+          Arguments to pass to pytest
+        type: string
```
[the remark this answers is not in the document]
Comment (amartsinovich-dialpad) [reply tester/orb.yaml:149]: hmm, technically arguments placed at the end of the command are passed to services' `ENDPOINT`, but you absolutely right, it's not necessarily `pytest`

File: tester/orb.yaml [hunk 3]
```diff
@@ -134,3 +134,47 @@ jobs:
           command: |
             cd <<parameters.cwd>>
             python -m pytest <<parameters.args>>
+
+  integration-tests:
```
Comment (TheKevJames) [inline tester/orb.yaml:138]: Let's call this `docker-compose`, I think, to go alongside the `pytest` job and avoid any assumptions here -- it's not necessarily the case that docker-compose is used only for integration tests, it could just as well get used for unit tests etc in the future.

File: tester/orb.yaml [hunk 4]
```diff
@@ -134,3 +134,47 @@ jobs:
...
+    description:
+      Runs integration tests using docker-compose
```
Comment (TheKevJames) [inline tester/orb.yaml:140]: Ditto re nuking "integration tests" from these descriptions

File: tester/orb.yaml [hunk 5]
```diff
@@ -134,3 +134,47 @@ jobs:
...
+      executor:
+        default: cimg/base:2020.01
+        description: >
+          Name of the docker image to use to execute the job.
+        type: string
+      resource_class:
+        default: small
+        type: string
+      service_name:
+        default: integration-tests
```
Comment (TheKevJames) [inline tester/orb.yaml:164]: Let's make this a required arg instead of having a default value here -- this is a bit more opinionated than we should be in an OSS context :)

File: tester/orb.yaml [hunk 6]
```diff
@@ -134,3 +134,47 @@ jobs:
...
+    steps:
+      - setup_remote_docker:
+          version: 18.09.3
+      - checkout
+      - run:
+          working_directory: <<parameters.cwd>>
+          # has to be built explicitly https://github.com/docker/compose/issues/7336
+          command: |
+            docker-compose build <<parameters.service_name>>
+            docker-compose run --rm <<parameters.service_name>> <<parameters.args>>
```
Comment (TheKevJames) [inline tester/orb.yaml:177]: nit: probably better to move the `run` command into its own CI step, since Circle will visualize and analyze steps independently -- by moving the `run` step into an explicit second section we'll be able to see at a glance whether "building" or "running" failed rather than needing to dive deeper into logs for that surface information.
